test/DynamoCoreTests/NodeToCodeTest.cs

```diff
@@ -1629,7 +1629,7 @@ protected void MutationTest(string dynFilePath)
 
             RunModel(dynFilePath);
             // Block until all tasks are executed
-            while (CurrentDynamoModel.Scheduler.HasPendingTasks) ;
+            while (CurrentDynamoModel.Scheduler.HasPendingTasks || CurrentDynamoModel.Scheduler.HasTaskInProgress);
```

Comment:
Was there any specific reason to add the `HasTaskInProgress` flag? Is the absence of it causing any issues where the check for pending tasks is insufficient? It appears that checking for this new flag will ensure that the last task (task queue should be empty when the last task is executing) also finishes, correct?

Reply:
Yes

Reply:
Can you elaborate what kind of issues you're seeing without this flag? Curious to know.

Reply:
It is exactly like you described it so basically, we were trying to retrieve some information too early before a task was completed. 
The issue was always there just that in Revit context we are relying on OnIdle thread to execute/interrogate DynamoModel and workspace and since we are in the same process. The OnIddle ensured there was nothing left to do since we are in the same process. 
Now in CLIMode we might want to do these checks from another thread so need to check to make sure all tasks are finished before we perform an interrogation.